### PR TITLE
fix(elasticsearch): advancedOptions in domain has no effect

### LINKED ISSUE #14067: (elasticsearch): DomainProps.advancedOptions has no effect (doesn't actually set AdvancedOptions)
DomainProps has an 'advancedOptions' field [[code](https://github.com/aws/aws-cdk/blob/63923ff431afdc031ee27a0fc9a99ffe4eb14405/packages/%40aws-cdk/aws-elasticsearch/lib/domain.ts#L407-L412)] [[doc](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-elasticsearch.DomainProps.html#advancedoptions)]:

```
  /**
   * Additional options to specify for the Amazon ES domain.
   *
   * @default - no advanced options are specified
   */
  readonly advancedOptions?: { [key: string]: (string) };
```

However, this field doesn't seem to be used anywhere. [[code search](https://github.com/aws/aws-cdk/search?q=advancedOptions)]

In particular, it's not used to set the 'AdvancedOptions' field in CloudFormation's AWS::Elasticsearch::Domain resource [[doc](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticsearch-domain.html)], which was surely the intent.

### Reproduction Steps

- Take any CDK code that uses the 'Domain' class to create an Elasticsearch domain.
- Add the 'advancedOptions' field, which is a map from string keys to string values, containing whatever mappings strike your fancy.
    - The "Advanced Options" section at https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-createupdatedomains.html#es-createdomain-configure-advanced-options lists the ones that Amazon Elasticsearch Service currently supports, though I think that other mappings might still be accepted as no-ops.
- Execute the CDK code to obtain the generated CloudFormation template.
- [optional] Deploy the CloudFormation template to create the actual Elasticsearch domain.

### What did you expect to happen?

- The generated CloudFormation template should include an 'AdvancedOptions' field with the specified mappings.
- The resultant Elasticsearch domain should have the specified advanced options.

### What actually happened?

- The generated CloudFormation template does *not* have an 'AdvancedOptions' field, and more generally, the 'advancedOptions' field seems to have no effect on the generated CloudFormation template. (I confirmed this via `diff -r`.)
- The resultant Elasticsearch domain does *not* have the specified advanced options, and more generally, the 'advancedOptions' field has no effect on the resultant Elasticsearch domain.

### Environment

- cdk 1.96.0
- TypeScript ~4.0.5

### Other

N/A

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by jumic**
Property `advancedOptions` in ElasticSearch domain did have no effect because the assignment was missing.
* add assignment for advancedOptions to fix issue
* test cases
* describe function in readme

Fixes #14067

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by peterwoodworth — DISMISSED** [review]
This PR looks great! Thank you for fixing this oversight

File: packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts [hunk 1]
```diff
@@ -1645,6 +1645,34 @@ describe('unsigned basic auth', () => {
   });
 });
 
+describe('advanded options', () => {
```
Comment (BenChaimberg) [inline packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:1648]:
```suggestion
describe('advanced options', () => {
```

File: packages/@aws-cdk/aws-elasticsearch/README.md [hunk 2]
```diff
@@ -290,3 +290,16 @@ new Domain(stack, 'Domain', {
...
+
+## Advanced options
+
+[Advanced options](https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-createupdatedomains.html#es-createdomain-configure-advanced-options) can used to configure additional options.
+
+```ts
+new Domain(stack, 'Domain', {
+    version: ElasticsearchVersion.V7_7,
+    advancedOptions: {
+        'rest.action.multi.allow_explicit_index': 'false',
```
Comment (BenChaimberg) [inline packages/@aws-cdk/aws-elasticsearch/README.md:302]: let's put all three documented options here

**Review by BenChaimberg — CHANGES_REQUESTED** [review]
Can you please also add one or more of these to the integ.elasticsearch.ts integration test?

File: packages/@aws-cdk/aws-elasticsearch/README.md (hunk 2 above)
Comment (jumic) [reply packages/@aws-cdk/aws-elasticsearch/README.md:302]: done

**Comment by jumic** [comment]
I added the integration test as requested and changed the readme.

I also hat to change the policy in the integration test. I received the message 'Apply a restrictive access policy to your domain' when I deployed the stack. My changes where not active at this time. I replaced `iam.AnyPrincipal()`by `iam.AccountRootPrincipal()` to use a valid policy.

**Review by BenChaimberg — APPROVED** [review]

**Pull request merged**